### PR TITLE
Clean up esoil

### LINKED ISSUE #765: Cleanup: make ET partitioning more explicit
- Version: VIC 5.0.1 develop branch
- Problem: the logic in computing soil evaporation (esoil) is not very clean nor does it make clear which portion of ET comes from esoil, until the partitioning is cleared up in put_data().
- Another problem: the default value of RARC_SOIL (architectural resistance for soil evaporation) is currently set to 100 s/m.  Evaluation of VIC performance at Ameriflux towers suggests a more optimal value would be 250 s/m.
- Solution: (a) introduce explicit `esoil`, `transp`, and `evap` terms into the `layer` struct; (b) remove the `bare_evap_frac` term; (c) modify logic in `func_surf_energy_bal()` and other functions to use these terms; (d) make the logic cleaner; and (e) change the default value of `RARC_SOIL` to 250.0 s/m.

### PR DESCRIPTION
**Pull request by tbohn**
- [x] closes #765
- [x] tests passed
- [ ] new tests added
- [ ] science test figures
- [ ] ran uncrustify prior to final commit
- [x] ReleaseNotes entry

Description (pasted from issue #765):

- Problem: the logic in computing soil evaporation (esoil) is not very clean nor does it make clear which portion of ET comes from esoil, until the partitioning is cleared up in put_data().
- Another problem: the default value of RARC_SOIL (architectural resistance for soil evaporation) is currently set to 100 s/m. Evaluation of VIC performance at Ameriflux towers suggests a more optimal value would be 250 s/m.
- Solution: (a) introduce explicit esoil, transp, and evap terms into the layer struct; (b) remove the bare_evap_frac term; (c) modify logic in func_surf_energy_bal() and other functions to use these terms; (d) make the logic cleaner; and (e) change the default value of RARC_SOIL to 250.0 s/m.
- Note: I've since rolled back the change to RARC_SOIL, due to this perhaps not being the best way to address the problem (and it being of a different nature than a simple code clean-up).

Note: this PR includes the changes from PR #723.  So the diff shows both what's in #723 and what's new to this PR.  Probably ought to wait until #723 is resolved  before reviewing this one.

### CODE REVIEW

File: vic/drivers/shared_all/src/initialize_parameters.c [hunk 1]
```diff
@@ -55,7 +55,7 @@ initialize_parameters()
     param.EMISS_H2O = 0.98;
 
     // Soil Constraints
-    param.SOIL_RARC = 100.0;
```
Comment (jhamman) [inline vic/drivers/shared_all/src/initialize_parameters.c]: can you explain this change?

Comment (tbohn) [reply vic/drivers/shared_all/src/initialize_parameters.c]: I've done some more comprehensive evaluations of VIC at the Ameriflux towers and found that 250 s/m is actually a more optimal value than 100.

Comment (jhamman) [reply vic/drivers/shared_all/src/initialize_parameters.c]: That's fine but this change should probably go in a separate PR.

Comment (tbohn) [reply vic/drivers/shared_all/src/initialize_parameters.c]: OK, I've broken out the `RARC_SOIL` change into a separate issue (#767) and will back out that change from this PR.  I'll make a new PR for the `RARC_SOIL` change.

**Comment by jhamman** [comment]
Okay, I'm going to merge this. I'll do a final uncrustify before 5.1.

**Comment by jhamman** [comment]
This was merged prematurely. I meant to merge #723. This is being reverted in #772.

**Pull request merged**